packages/react/src/components/modal/dialog/modal-dialog.tsx

```diff
@@ -0,0 +1,144 @@
+import { Fragment, ReactElement, Ref, useMemo, useRef, VoidFunctionComponent } from 'react';
+import { useTranslation } from '../../../i18n/use-translation';
+import { v4 as uuid } from '../../../utils/uuid';
+import { useDeviceContext } from '../../device-context-provider/device-context-provider';
+import { Heading } from '../../heading/heading';
+import { Modal } from '../modal';
+import {
+    ButtonContainer,
+    CancelButton,
+    ConfirmButton,
+    StyledHeadingWrapperComponent,
+    Subtitle,
+    TitleIcon,
+} from './styled';
+import { DialogType, ModalDialogProps } from './types';
+
+const ModalRoles: Record<DialogType, string> = {
+    information: 'dialog',
+    action: 'dialog',
+    alert: 'alertdialog',
+};
+
+export const ModalDialog: VoidFunctionComponent<ModalDialogProps> = ({
+    appElement,
+    ariaDescribedby,
+    ariaHideApp,
+    dialogType = 'action',
+    cancelButton,
+    children,
+    className,
+    confirmButton,
+    footerContent,
+    hasCloseButton,
+    isOpen,
+    width,
+    parentSelector,
+    shouldCloseOnOverlayClick = true,
+    subtitle,
+    title,
+    titleIcon,
+    onRequestClose,
+}) => {
+    const { isMobile } = useDeviceContext();
+    const { t } = useTranslation('modal-dialog');
+    const titleId = useMemo(uuid, []);
```

Comment:
```suggestion
    const titleId = useId();
```
https://github.com/kronostechnologies/design-elements/blob/master/packages/react/src/hooks/use-id.ts